### ISSUE TITLE
CSGO's VertexlitGeneric fixes and additions.

### ISSUE THREAD

**Issue by One-Trick-Viper** [Facepunch/garrysmod-requests#1839]
Seeing as something like completely new shaders are entirely out of the question, I was wondering if there is any possibility to get CSGO's VertexlitGeneric fixes and additions?

A fix I mainly want to see implemented is how CSGO's engine branch handles the $phongexponenttexture:

CSGO's engine branch:
![image](https://user-images.githubusercontent.com/29631812/112767015-b710a880-9014-11eb-86d9-b6badb2e8ca1.png)

GMod's:
![image](https://user-images.githubusercontent.com/29631812/112767022-baa42f80-9014-11eb-8b54-6ff3c3a8ddbd.png)

$phongalbedoboost would also be a greatly appreciated feature.

Something that CSGO doesn't fix but that would be great if it were to be fixed is that $envmapsaturation and $envmapcontrast do not work when $phong is enabled.

**Comment by robotboy655** [comment]
Partial or complete duplicate of https://github.com/Facepunch/garrysmod-requests/issues/939

**Comment by One-Trick-Viper** [comment]
> 4 years ? clearly, S&Box will come out faster than we will wait for the support of modern shaders. 🤦‍♂️

dont worry, dont think I actually expect this to come at all. I just posted it here for the sake of my own sanity knowing it;'s tucked away somewhere and labelled at least..

**Comment by FlorianLeChat** [comment]
> As soon as at least some S&Box beta is released to which I will have access, I immediately abandon this outdated engine and start studying Source 2. I'm already tired of enduring these terrible restrictions. A lot of my ideas are crashing about engine capability and Rubat laziness.

I feel sorry for the S&Box developers who will have to deal with people like you in few months.

**Comment by MARK2580** [comment]
> > As soon as at least some S&Box beta is released to which I will have access, I immediately abandon this outdated engine and start studying Source 2. I'm already tired of enduring these terrible restrictions. A lot of my ideas are crashing about engine capability and Rubat laziness.
> 
> I feel sorry for the S&Box developers who will have to deal with people like you in few months.

They forgot to ask you.

**Comment by FlorianLeChat** [comment]
> They forgot to ask you. I am enraged by all sorts of coders who think that the main thing in this game is the lua code.

If you're mad just because people prefer to code things instead of making maps, I feel sorry for you, lol. Also, most experienced coders are dependent on other things than just "Lua code" to create addons, like creating maps or modeling stuff.

> And the fact that the game does not support new shaders, does not support a bunch of new functions in models, there are still no lightmaps on models, bump still disables vertex lighting on models. It is even banal to create a x4 blending for displacement !!!
This problem with phong has already turned out to be 4 years old and in 4 years it really was impossible to do anything, I'm just in shock.

What did you expect to do on a game that is more than 15 years old, honestly? It is normal that many features are missing or buggy. Obviously, Garry's Mod has a lot of important improvements compared to its beginnings (or to the Half-Life 2 game it inherited) but you can't reinvent the wheel.

Also, your criticism towards Rubat is totally unjustified considering the work he must do to manage everything at the same time. The support on Discord, the wiki stuff, the management and handling of new bugs/requests and the introduction of new features. And guess what? He's terribly alone. To say he's a slacker doesn't make me think that, and considering the changes he's made in the last few years, I don't think so.  He's not a god, just a person who tries to keep a game up to date, fix bugs, fix requests and implement new features if possible.

You are constantly complaining about the slowness in adding some new Source game features but considering the resources Facepunch invested in the game, I find the result quite satisfying for my part. And if Rubat thinks that this feature is not necessary and/or will never be implemented, it's because he knows that this time could be invested in solving the thousands of requests still pending on Github.

Finally, you are not alone in asking for more features, many are waiting for the official release of the 64-bit version of the game, the replacement of the Awesomium renderer, new functions, hooks, libraries, and more. But, we're not constantly harassing Rubat to do it because we are for the majority, comprehensive.

**Comment by MARK2580** [comment]
@FlorianLeChat
By the way, there are only some problems with the Awesomium library, recently the first launch of the game takes just a hell of a lot of time, the game can start up to the main menu for about a minute, even the most difficult modern games like GTA V or RDR2 do it much faster.

I know perfectly well that Source 1 is totally outdated and I have mentioned it more than once, but I see that a CSGO build can and I want to at least see such a level of capabilities, it is banal to see phong on brushes, this will radically change the appearance of the maps.

But it looks like I would rather wait for the public test of S&Box than in gmod there will be opportunities from CSGO.
Basically everything that interests me is based on modern shaders. When I recreate a map from another game, every time I run into the limited build gmod, I could have done much more and more beautiful on the same CSGO, but only mentally retarded schoolchildren play CSGO, and Gmod is a sandbox where you can do what whatever.
At the moment, creating maps for Gmod is my bread in the truest sense of the word. I do it for a living.

**Comment by One-Trick-Viper** [comment]
man i just wanted csgo phong what is this shit

**Comment by MARK2580** [comment]
> man i just wanted csgo phong what is this shit

FlorianLeChat just came in and started it all. I think it's time to bury Gmod, it has served its time. We are waiting for the release of S&Box.

**Comment by CaskoChan** [comment]
Just gonna leave this here https://github.com/Facepunch/garrysmod-requests/issues/1195

**Comment by One-Trick-Viper** [comment]
> Just gonna leave this here #1195

I highly doubt we'll see any of this in gmod, ever. but it's worth a try I guess.
